app.js

```diff
@@ -2,12 +2,27 @@ const express = require("express");
 const app = express();
 const { getAllTopics } = require("./controllers/topics.controller");
 const { getApi } = require("./controllers/api.controller");
+const { getArticleById } = require("./controllers/articles.controller");
 
 app.get("/api/topics", getAllTopics);
 app.get("/api", getApi);
+app.get("/api/articles/:article_id", getArticleById);
 
 app.all("*", (req, res) => {
   res.status(404).send({ msg: "Not found" });
 });
 
+app.use((err, req, res, next) => {
+  if (err.code) {
+    res.status(400).send({ msg: "Bad request" });
+  } else {
+    next;
+  }
+});
```

Comment:
Nicely implemented custom error handler - but make sure to invoke next with the err object here, to make sure it reaches the next error handler!